Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -59,12 +59,12 @@ std::optional<TimeZoneID> parseTimeZoneName(StringView string)
 }
 
 template<typename CharType>
-static int32_t parseDecimalInt32(const CharType* characters, unsigned length)
+static int32_t parseDecimalInt32(std::span<const CharType> characters)
 {
     int32_t result = 0;
-    for (unsigned index = 0; index < length; ++index) {
-        ASSERT(isASCIIDigit(characters[index]));
-        result = (result * 10) + characters[index] - '0';
+    for (auto character : characters) {
+        ASSERT(isASCIIDigit(character));
+        result = (result * 10) + character - '0';
```

Comment:
Seems like this just overflows. Is that really what we want? I guess callers always pass only short strings here. Surprised there isn’t a function elsewhere we can reuse.